spec/system/admin/webhooks_spec.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Admin Webhooks' do
+  describe 'Managing webhooks' do
+    before { sign_in Fabricate(:user, role: UserRole.find_by(name: 'Admin')) }
+
+    describe 'Viewing webhooks' do
+      let!(:webhook) { Fabricate :webhook }
+
+      it 'lists existing records' do
+        visit admin_webhooks_path
+
+        expect(page)
+          .to have_content(I18n.t('admin.webhooks.title'))
+          .and have_content(webhook.url)
+
+        click_on(webhook.url)
+        expect(page)
+          .to have_content(I18n.t('admin.webhooks.title'))
+      end
+    end
+
+    describe 'Creating a new webhook' do
+      it 'creates new record with valid attributes' do
+        visit admin_webhooks_path
+
+        click_on I18n.t('admin.webhooks.add_new')
+        expect(page)
+          .to have_content(I18n.t('admin.webhooks.new'))
+
+        fill_in 'webhook_url', with: ''
+        expect { submit_form }
+          .to_not change(Webhook, :count)
+        expect(page)
+          .to have_content(/errors below/)
```

Comment:
It's a minor thing, but we're not checking empty events anywhere.

Reply:
Oh yeah, it's testing empty URL and empty events at the same time. Maybe it should be split in two, and made a little more explicit with comments.

Reply:
Added some scattered comments to add some context about what's being exercised in a few spots.

Reply:
Will do ... but out of curiosity (mostly to inform future PRs along these lines) ... why?

Generally speaking I'd try to be exhaustive on all combinations of how we can make a thing invalid at the model spec level; and then at any of the controller/system/request spec level, just make sure we hit the "invalid path" once to assert that some errors showed up and the invalid attrs did not save a record (exception here would be if there were multiple ways to be invalid, multiple filters, other edge cases, etc - to exercise them all) ... but in a way where it sort of doesnt matter which particular combination of making it invalid we used.

Again, no objection, just want to understand rationale for future of these. Will update this.

Reply:
Added special case for no events selected.

Reply:
Yeah, I suppose you're right, if the model tests cover this, it does not matter much.